### PR TITLE
feat(web): license UI

### PR DESCRIPTION
**Pull request by alextran1502**

### CODE REVIEW

File: web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte [hunk 1]
```diff
@@ -28,15 +29,18 @@
...
   const logOut = async () => {
     const { redirectUri } = await logout();
+
     if (redirectUri.startsWith('/')) {
       await goto(redirectUri);
     } else {
       window.location.href = redirectUri;
     }
     resetSavedUser();
+    setLicenseStatus(false);
```
Comment (jrasm91) [inline web/src/lib/components/shared-components/navigation-bar/navigation-bar.svelte:43]: This should be inside of `resetSavedUser()`

**Review by danieldietzler — APPROVED** [review]

File: web/src/lib/components/shared-components/side-bar/license-info.svelte [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+</script>
+
+{#if isOpen}
+  <LicenseModal onClose={() => (isOpen = false)} />
+{/if}
+
+<div class="hidden md:block license-status pl-4 text-sm">
+  {#if $isLicenseActivated}
+    <button
+      on:click={() => goto(`${AppRoute.USER_SETTINGS}?isOpen=user-license-settings`)}
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/license-info.svelte:31]: This should just be a regular link

File: web/src/lib/components/shared-components/side-bar/license-info.svelte [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      <div class="flex gap-1 mt-2 place-items-center dark:bg-immich-dark-primary/10 bg-gray-100 py-3 px-2 rounded-lg">
+        <Icon path={mdiLicense} size="18" class="text-immich-primary dark:text-immich-dark-primary" />
+        <p class="dark:text-gray-100">{$t('license_info_licensed')}</p>
+      </div>
+    </button>
+  {:else}
+    <button
+      type="button"
+      on:click={openLicenseModal}
+      on:mouseenter={() => (showMessage = true)}
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/license-info.svelte:44]: The message that stays visible when I hover over this is pretty annoying

File: web/src/lib/components/shared-components/side-bar/license-info.svelte [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        </span>
+      </div>
+    </button>
+  {/if}
+</div>
+
+<Portal target="body">
+  {#if showMessage && getAccountAge() > 14}
+    <div
+      class="w-[265px] absolute bottom-[75px] left-[255px] bg-white dark:bg-gray-800 dark:text-white text-black rounded-xl z-10 shadow-2xl px-4 py-5"
```
Comment (michelheusschen) [inline web/src/lib/components/shared-components/side-bar/license-info.svelte:66]: The position of this message will be wrong when not using 100% zoom

**Review by michelheusschen — COMMENTED** [review]
Good job on the side bar design, I really like it!

**Comment by LucaDev** [comment]
I really hope this will not get annoying for unlicensed users. It would probably end in a fork of Immich with all of this removed. Calling it a „trial“ when no license is activated gives me WinRAR vibes.

**Comment by alextran1502** [comment]
@LucaDev the only difference is just that status text to show licensed vs unlicensed

**Comment by alextran1502** [comment]
@LucaDev For sure, we are not planning on that 😄

**Comment by Rigo-V** [comment]
The wording choice is a bit suspicious to me. Seeing as this is AGPL licenced and distributed freely, what does it mean when it says unpaid use is "unlicensed" as in "not permitted"? Even though the actual legal text of the license says I have the right to do *more* than use it for any number of users. Does this imply that the licence will be switched in the future to something that contains further restrictions on the use and/or modification of Immich?

If not, why not say something like "unpaid" instead then? To be clear, I have nothing against you making money with this, you fully deserve it. I love Immich so far, and would gladly pay you $25. I just fear it'll turn into some "source available" not-actually-free (as in freedom) software, and I'll have to migrate to something else (out of principle if nothing else).

**Comment by bo0tzz** [comment]
This PR is not the place for that discussion, please keep it to the announcement threads: #11186 and #11226. All your points and concerns have already been covered in there.

**Pull request merged**